src/Mobs/SnowGolem.cpp

```diff
@@ -56,3 +57,32 @@ void cSnowGolem::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 		}
 	}
 }
+
+
+
+
+
+bool cSnowGolem::Attack(std::chrono::milliseconds a_Dt)
+{
+	// Comment inherited from skeletons
+	StopMovingToPosition();  // Todo handle this in a better way, the snowman does some uneeded recalcs due to inStateChasing
+	auto & Random = GetRandomProvider();
+
+	if ((GetTarget() != nullptr) && (m_AttackCoolDownTicksLeft == 0))
+	{
+		Vector3d Inaccuracy = Vector3d(Random.RandReal<double>(-0.25, 0.25), Random.RandReal<double>(-0.25, 0.25), Random.RandReal<double>(-0.25, 0.25));
+		Vector3d Speed = (GetTarget()->GetPosition() + Inaccuracy - GetPosition()) * 4;
+		Speed.y += Random.RandInt(-1, 1);
```

Comment:
I've just decreased the speed. It makes it a little bit better for spider and not significantly worse for higher mobs. I'll see whether this can be improved by any simple change to the code calculating `Vector3d Speed`

Reply:
Well, after some analysis, I think what may be causing the code to perform so poorly for shorter mobs. The snowball is created at the snowman's head, but its speed is calculated using the snowman's feet position (simple `GetPosition()`). 

For higher mobs, this may just cancel out. You see - we also use target's feet position. This doesn't sound like a good idea, but...

We're calculating the snowball's speed like it's thrown at feet and is aiming at feet... but we're actually launching it higher. So it lands higher - at the chest/head, for average mobs, depending on the inaccuracy. But spiders are basically flat, their head are at the same height as their feet.

I think I'll find some time to tune the code today. Especially now, when I know what might be wrong

Reply:
Fixed